### PR TITLE
NavigationView: Close pane in non Left displaymode when selecting an item from AutoSuggestBox

### LINKED ISSUE #3241: When you go to a sample page from the search, the navigationview should close (if it is in compact mode)
Repro:
1. Open Xaml Controls Gallery so primary navigation is in LeftCompact mode
2. Search for Button and chose this page from the ASB dropdown

Actual behavior: XCG navigates to another page but NavigationView pane remains open.
Expected: XCG navigates and also closes the pane.

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Listening to QuerySubmitted event of AutoSuggestBox and closes pane when not in DisplayMode left
## Motivation and Context

Closes #3241
## How Has This Been Tested?

Add interaction test.

### CODE REVIEW

File: dev/NavigationView/NavigationView.cpp [hunk 1]
```diff
@@ -3789,6 +3791,17 @@ void NavigationView::OnPropertyChanged(const winrt::DependencyPropertyChangedEve
     else if (property == s_AutoSuggestBoxProperty)
     {
         InvalidateTopNavPrimaryLayout();
+        if (args.OldValue())
+        {
+            m_autoSuggestBoxQuerySubmittedRevoker.revoke();
+        }
+        if (const auto newSuggestBox = args.NewValue())
+        {
+            if (const auto autoSuggestBox = newSuggestBox.try_as<winrt::AutoSuggestBox>())
```
Comment (Felix-Dev) [inline dev/NavigationView/NavigationView.cpp:3800]: why not `args.NewValue().try_as<winrt::AutoSuggestBox>()`?

Comment (chingucoding) [reply dev/NavigationView/NavigationView.cpp:3800]: What if args.NewValue() is nullptr?

Comment (Felix-Dev) [reply dev/NavigationView/NavigationView.cpp:3800]: try_as should cover that case

Comment (chingucoding) [reply dev/NavigationView/NavigationView.cpp:3800]: Yep you are right, updated that.

File: dev/NavigationView/NavigationView_InteractionTests/PaneBehaviorTests.cs [hunk 2]
```diff
@@ -951,5 +951,61 @@ public void EnsureDynamicSizeForPaneHeaderFooterAndCustomContent()
...
+            {
+                var configs = new ControlWidth[] { ControlWidth.Narrow, ControlWidth.Medium };
+
+                foreach(var width in configs)
+                {
+                    SetNavViewWidth(width);
+
+                    CheckBox isPaneOpenCheckBox = new CheckBox(FindElement.ById("IsPaneOpenCheckBox"));
+
+                    // On phone, the pane will initially be in the closed compact state, so open it before
```
Comment (StephenLPeters) [inline dev/NavigationView/NavigationView_InteractionTests/PaneBehaviorTests.cs:969]: >phone [](start = 26, length = 5)

is this copied from somewhere? we don't test on phone anymore.

Comment (chingucoding) [reply dev/NavigationView/NavigationView_InteractionTests/PaneBehaviorTests.cs]: Yes I copied the skeleton of another test, updated the comment.

**Comment by StephenLPeters** [comment]
@chingucoding this looks like a real test failure to me

File: dev/NavigationView/NavigationView.cpp [hunk 3]
```diff
@@ -4515,6 +4525,15 @@ void NavigationView::UpdateTitleBarPadding()
     }
 }
 
+void NavigationView::OnAutoSuggestBoxQuerySubmitted(const winrt::AutoSuggestBox& sender, const winrt::Windows::UI::Xaml::Controls::AutoSuggestBoxQuerySubmittedEventArgs& args)
```
Comment (ranjeshj) [inline dev/NavigationView/NavigationView.cpp:4528]: would we get this event if there were no matches for the query ? If there is no selection in that case, perhaps we should keep the pane open.

Comment (chingucoding) [reply dev/NavigationView/NavigationView.cpp:4528]: That's a very good point. I think the SuggestionChosen event is more appropriate in this case, do you agree?

Comment (ranjeshj) [reply dev/NavigationView/NavigationView.cpp:4528]: That might be a better event. There is also an args on this event that has some suggestions, but I don't think that guarantees a selection.

**Comment by chingucoding** [comment]
@StephenLPeters Test failure should be fixed now. @ranjeshj I added a check to ensure we only close the pane if the selected suggestion is not null, if there are any precautions we should take there please let me know.

**Comment by StephenLPeters** [comment]
@chingucoding can you address the merge conflict as well?

**Comment by chingucoding** [comment]
Oh right, totally missed. Merged with master now.

**Comment by StephenLPeters** [comment]
Looks like an unlucky unreliable tests.

**Review by StephenLPeters — APPROVED** [review]
:shipit:

**Pull request merged**